packages/api/src/command/hie/unlink-patient-from-organization.ts

```diff
@@ -0,0 +1,345 @@
+      errors.push({ documentId: document.id, error });
+      continue;
+    }
+  }
+
+  if (errors.length > 0) {
+    capture.error("Failed to process some documents during unlink", { extra: { errors } });
+  }
+
+  log(`Completed unlinking patient from organization`);
+}
+
+function findCwLinkWithOid(cwPatientData: CwData | undefined, oid: string): CwLink | undefined {
+  if (!cwPatientData) return undefined;
+
+  const cwLinks = cwPatientData.links;
+
+  for (const cwLink of cwLinks) {
+    const patient = cwLink.patient;
+    if (!patient) continue;
+
+    if (patient.identifier?.some(identifier => identifier.system === addOidPrefix(oid))) {
+      return cwLink;
+    }
+  }
+
+  return undefined;
+}
+
+function findCqLinkWithOid(cqPatientData: CQData | undefined, oid: string): CQLink | undefined {
+  if (!cqPatientData) return undefined;
+
+  const cqLinks = cqPatientData.links;
+
+  return cqLinks.find(link => link.oid === oid);
+}
+
+function getDocumentsWithOid(
+  documents: DocumentReferenceWithId[],
+  oid: string
+): DocumentReferenceWithId[] {
+  const urnOid = addOidPrefix(oid);
+  const commonwellDocuments = documents.filter(hasCommonwellExtension);
+  const carequalityDocuments = documents.filter(hasCarequalityExtension);
+
+  const matchingDocumentRefs = [];
+
+  for (const document of commonwellDocuments) {
+    const patient = document.contained?.find(isPatient);
+    if (!patient) continue;
+
+    const identifier = patient.identifier?.find(identifier => identifier.system === urnOid);
+    if (identifier) {
+      matchingDocumentRefs.push(document);
+    }
+  }
+
+  for (const document of carequalityDocuments) {
+    const organization = document.contained?.find(isOrganization);
+    if (!organization) continue;
+
+    const identifier = organization.identifier?.find(identifier => identifier.value === oid);
+
+    if (identifier) {
+      matchingDocumentRefs.push(document);
+    }
+  }
+
+  return matchingDocumentRefs;
+}
+
+function getS3FileNameFromDocument(
+  document: DocumentReferenceWithId,
+  log: typeof console.log
+): string | undefined {
+  const s3Attachment = getMetriportContent(document)?.attachment;
+  const fileName = s3Attachment?.title;
+
+  if (!fileName) {
+    log(`No file name found in document ${document.id}`);
+    return undefined;
+  }
+
+  return fileName;
+}
+
+async function findAndRemoveConversionResultsFromS3(
+  fileName: string,
+  dryRun = false,
+  log: typeof console.log
+): Promise<void> {
+  const dryRunMsg = getDryRunPrefix(dryRun);
+  try {
+    const objects = await s3Utils.listObjects(s3ConversionResultBucketName, fileName);
+    if (!objects) return;
+
+    const validFiles = objects.flatMap(obj => obj.Key ?? []);
+    if (validFiles.length === 0) return;
+
+    log(`${dryRunMsg}Deleting ${validFiles.length} files from S3`);
+
+    if (!dryRun) {
+      await s3Utils.deleteFiles({
+        bucket: s3ConversionResultBucketName,
+        keys: validFiles,
+      });
+    }
```

Comment:
Can we add a log indicating it succeeded, please? I'm thinking that, if one of the promises in the `Promise.all` fail, we'd want to know which ones succeeded, since this would be non-deterministic as they happen in parallel.

If this makes sense, add it here and other related functions called by the same `Promise.all` please.